src/utils/EventUtils.ts

```diff
@@ -1,6 +1,11 @@
 import * as React from 'react';
 import {isEmpty} from 'underscore';
 
+export interface IConditionalOnClickCallBack {
+    callBack: (event?: any, data?: any) => void;

```

Comment:
if you have the data in the callback, why create a condition too with the same argument? What i'm trying to say is you could get away with only the callback and write a condition inside it if you need to, no need for the `condition` attribute/property

Reply:
Yep, its a good point but I want to stay with the same idea than the additionalClass. I think its more clear to have the condition in the table props instead of inside each callBack. You can handle only the callback behaviour without the condition eveytime